### PR TITLE
add getMinimalUpperBounds to Poset

### PR DESCRIPTION
**Pull request by dwightguth**
@yilongli please review

### CODE REVIEW

**Comment by yilongli** [comment]
The refactoring is pretty nice. Thank you, Dwight. Merge if all tests pass.

**Comment by yilongli** [comment]
After re-review, I found some inconsistencies regarding the naming of variables and comments due to the refactoring extracting common functionality. Could you please also update them? Thanks.

**Comment by dwightguth** [comment]
@yilongli please re-review

File: src/javasources/KTool/src/org/kframework/utils/Poset.java [hunk 1]
```diff
@@ -154,25 +125,17 @@ public String getLUB(Set<String> subset) {
      * 
      */
     public String getGLB(Set<String> subset) {
+        return getUniqueBound(subset, lowerBound);
+    }
+    
+    private String getUniqueBound(Set<String> subset, BoundType type) {
```
Comment (andreistefanescu) [inline src/javasources/KTool/src/org/kframework/utils/Poset.java:131]: Could you implement this using `getClosestBounds` and checking the result has just one element?

**Comment by yilongli** [comment]
lgtm except the one last minor thing I just commented

File: src/javasources/KTool/src/org/kframework/utils/Poset.java (hunk 1 above)
Comment (yilongli) [inline src/javasources/KTool/src/org/kframework/utils/Poset.java:131]: this unique bound refers to GLB or LUB which is very likely not one of the closest bounds

**Comment by dwightguth** [comment]
@yilongli don't forget to make an issue for that refactoring we proposed.

**Comment by yilongli** [comment]
yep, done. https://github.com/kframework/k/issues/652

**Pull request merged**